pkg/icingadb/history/sync.go

```diff
@@ -32,274 +37,359 @@ func NewSync(db *icingadb.DB, redis *icingaredis.Client, logger *zap.SugaredLogg
 	}
 }
 
-// insertedMessage represents a just inserted row.
-type insertedMessage struct {
-	// redisId specifies the origin Redis message.
-	redisId string
-	// structType represents the table the row was inserted into.
-	structType reflect.Type
-}
+// redisCount is the number of history entries to read at once (using XREAD) or delete at once (using XDEL).
+const redisCount = 1 << 10
```

Comment:
We already have values for count `HSCAN` and `HMGET`. We could combine those and this const into just one count.

Reply:
I've added a new option for now. We could still unify this in a later PR, but I think the intention of these is to have some tuning knobs should the defaults be suboptimal. So it doesn't hurt to have then fine-grained in my opinion.